pkg/controller/controlplanecerts/controlplanecerts_controller.go

```diff
@@ -375,3 +387,28 @@ func controlPlaneCertsSyncSetName(name string) string {
 func defaultControlPlaneDomain(cd *hivev1.ClusterDeployment) string {
 	return fmt.Sprintf("api.%s.%s", cd.Spec.ClusterName, cd.Spec.BaseDomain)
 }
+
+func writeSecretData(w io.Writer, secret *corev1.Secret) {
+	// sort secret keys so we get a repeatable hash
+	keys := make([]string, 0, len(secret.Data))
+	for k := range secret.Data {
+		keys = append(keys, k)
+	}
+	sort.StringSlice(keys).Sort()
+	fmt.Fprintf(w, "%s/%s\n", secret.Namespace, secret.Name)
+	for _, k := range keys {
+		fmt.Fprintf(w, "%s: %x\n", k, secret.Data[k])
+	}
+}
+
+func secretsHash(secrets []*corev1.Secret) string {
```

Comment:
This func could probably use a unit test just to make sure we're getting consistent hashes, but then I'm not entirely sure if we can control the order of the secret keys in a test?